### DISCUSSION TITLE
Adding freecash support (Block submission failed with: bad-cb-no-2nd-out)

### DISCUSSION THREAD

**Discussion by peysab** [coinfoundry/miningcore#856] [Ideas]
I am trying to add FreeCash to the pool, but while submitting I am getting the following error:
Block 85599 submission failed with: bad-cb-no-2nd-out
this is the coin config file:
` "free-cash": {
        "name": "Free Cash",
        "symbol": "FCH",
        "family": "bitcoin",
        "coinbaseHasher": {
            "hash": "sha256d"
        },
        "headerHasher": {
            "hash": "sha256d"
        },
        "blockHasher": {
            "hash": "reverse",
            "args": [ { "hash": "sha256d" } ]
        },
        "explorerBlockLink": "https://freecash.info/block?height={0}",
        "explorerTxLink": "https://freecash.info/transaction?hash={0}",
        "explorerAccountLink": "https://freecash.info/address?address={0}"
    },`

and here is the config file:

`  {
            "id": "fch",
            "enabled": true,
            "coin": "free-cash",
            "poolName": "Free Cash",
            "address": "F6q4QiYXJtkis5hK4gwBHJy8WPQpMwGK86",
            "rewardRecipients": [
                {
                    "address": "FELwdc2YAmXsaeKZVD8JKaJtxe2UECPYNX",
                    "percentage": 0.1
                }
            ],
            "blockRefreshInterval": 250,
            "jobRebroadcastTimeout": 10,
            "clientConnectionTimeout": 600,
            "banning": {
                "enabled": false,
                "time": 30,
                "invalidPercent": 50,
                "checkThreshold": 50
            },
            "ports": {
                "55555": {
                    "listenAddress": "0.0.0.0",
                    "difficulty": 16000,
                    "name": "ASIC Mining",
                    "varDiff": {
                        "minDiff": 4096,
                        "maxDiff": 250000,
                        "targetTime": 10,
                        "retargetTime": 60,
                        "variancePercent": 20
                    }
                },
                "55556": {
                    "listenAddress": "0.0.0.0",
                    "difficulty": 500000,
                    "name": "High Diff (Nicehash)",
                    "varDiff": {
                        "minDiff": 500000,
                        "maxDiff": 1250000,
                        "targetTime": 10,
                        "retargetTime": 60,
                        "variancePercent": 20
                    }
                }
            },
            "daemons": [
                {
                    "host": "127.0.0.1",
                    "port": 61666,
                    "user": "xxxxx",
                    "password": "xxxxx",
                    "zmqBlockNotifySocket": "tcp://127.0.0.1:36014"
                }
            ],
            "paymentProcessing": {
                "enabled": true,
                "minersPayTxFees": true,
                "minimumPayment": 5,
                "payoutScheme": "PPLNS",
                "payoutSchemeConfig": {
                    "factor": 2
                }
            }
        },`

Any Ideas how to fix this issue?

**Comment by peysab** [comment]
@oliverw 
I checked coins source code and the condition to return that error is:

`if (tx.vout.size() < 2) {
        return state.DoS(100, false, REJECT_INVALID, "bad-cb-no-2nd-out");
    }`

So it means we need to set 2 vout for tx, but I couldn't find anywhere in the settings to do that.
can you help me please?

**Comment by peysab** [comment]
I also enabled both hasmasternode and haspayee but the outcome was same

**Comment by mrmikeo** [comment]
Try setting multiple reward recipients in config.json.

**Comment by akshaynexus** [comment]
that wouldnt work,need to check the coin code then

**Comment by belmix** [comment]
how to fix?

https://github.com/freecashorg/freecash/wiki/Freecash%E5%8C%BA%E5%9D%97%E7%BB%93%E6%9E%84-%E5%8F%8A-%E6%8C%96%E7%9F%BF%E5%8D%8F%E8%AE%AE%E6%94%B9%E8%BF%9B

"coinbasedevreward": {
    "value": 2500000000,
    "scriptpubkey": "76a914f1704a9663c1e530f82ca1bc7ff52f0f65abc1ca88ac"
  },

FTqiqAyXHnK7uDTXzMap3acvqADK4ZGzts - Developers address

**Comment by akshaynexus** [comment]
Someone needs to add coinbasedevreward support to freecash that's all

**Comment by akshaynexus** [comment]
Here,added FreeCash support
https://github.com/akshaynexus/miningcore/commit/9da45f2a6c59437527c362a6b088d2ede6f8b609

**Comment by calvintam236** [comment]
> Here,added FreeCash support
> [akshaynexus@9da45f2](https://github.com/akshaynexus/miningcore/commit/9da45f2a6c59437527c362a6b088d2ede6f8b609)

Please open a PR if you want to add freecash support into the main repo.
